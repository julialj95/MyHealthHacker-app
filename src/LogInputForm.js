import React from "react";

class LogInputForm extends React.Component {
  render() {
    return (
      <form onSubmit={this.props.handleSubmitLog}>
        <section>
          <label htmlFor="date">Current Date</label>
          <input
            type="date"
            id="log_date"
            name="log_date"
            value={this.state.log_date}
            onChange={this.props.handleChange}
          />
        </section>
        <br />

        <div className="log_section">
          <section className="mental">
            <h2>Psychological</h2>
            <div className="mood_rating">
              <h3>Mood Rating</h3>
              <h3>Rate your mood out of 5</h3>
              <input
                type="radio"
                id="1"
                name="mood"
                value={1}
                onClick={this.props.handleChange}
              />
              <label htmlFor="mood">1</label>
              <input
                type="radio"
                id="2"
                name="mood"
                value={2}
                onClick={this.props.handleChange}
              />
              <label htmlFor="mood">2</label>
              <input
                type="radio"
                id="3"
                name="mood"
                value={3}
                onClick={this.props.handleChange}
              />
              <label htmlFor="mood">3</label>
              <input
                type="radio"
                id="4"
                name="mood"
                value={4}
                onClick={this.props.handleChange}
              />
              <label htmlFor="mood">4</label>
              <input
                type="radio"
                id="5"
                name="mood"
                value={5}
                onClick={this.props.handleChange}
              />
              <label htmlFor="mood">5</label>
            </div>
            <div className="stress">
              <h2>Stress Level</h2>
              <h3>Rate your stress level with 5 being the most stressed</h3>
              <input
                type="radio"
                id="1"
                name="stress"
                value={1}
                onClick={this.props.handleChange}
              />
              <label htmlFor="stress">1</label>
              <input
                type="radio"
                id="2"
                name="stress"
                value={2}
                onClick={this.props.handleChange}
              />
              <label htmlFor="stress">2</label>
              <input
                type="radio"
                id="3"
                name="stress"
                value={3}
                onClick={this.props.handleChange}
              />
              <label htmlFor="stress">3</label>
              <input
                type="radio"
                id="4"
                name="stress"
                value={4}
                onClick={this.props.handleChange}
              />
              <label htmlFor="stress">4</label>
              <input
                type="radio"
                id="5"
                name="stress"
                value={5}
                onClick={this.props.handleChange}
              />
              <label htmlFor="stress">5</label>
            </div>
          </section>
          <section className="sleep">
            <h2>Sleep</h2>
            <label htmlFor="sleep">Hours slept</label>
            <br />
            <input
              type="text"
              name="sleep_hours"
              id="sleep_hours"
              value={this.state.sleep_hours}
              onChange={this.props.handleChange}
            />
            <div>
              <h3>Sleep quality</h3>
              <input
                type="radio"
                id="1"
                name="sleep_quality"
                value={1}
                onClick={this.props.handleChange}
              />
              <label htmlFor="sleep_quality">1</label>
              <input
                type="radio"
                id="2"
                name="sleep_quality"
                value={2}
                onClick={this.props.handleChange}
              />
              <label htmlFor="sleep_quality">2</label>
              <input
                type="radio"
                id="3"
                name="sleep_quality"
                value={3}
                onClick={this.props.handleChange}
              />
              <label htmlFor="sleep_quality">3</label>
              <input
                type="radio"
                id="4"
                name="sleep_quality"
                value={4}
                onClick={this.props.handleChange}
              />
              <label htmlFor="sleep_quality">4</label>
              <input
                type="radio"
                id="5"
                name="sleep_quality"
                value={5}
                onClick={this.props.handleChange}
              />
              <label htmlFor="sleep_quality">5</label>
            </div>
          </section>
          <br />
          <section className="exercise">
            <h2>Physical</h2>
            <label htmlFor="exercise">Minutes of exercise</label>
            <br />
            <input
              type="text"
              name="exercise"
              id="exercise"
              value={this.state.exercise}
              onChange={this.props.handleChange}
            />
            <br />
            <br />
            <label htmlFor="exercise_type">Type of exercise</label>
            <br />
            <input
              type="text"
              name="exercise_type"
              id="exercise_type"
              value={this.state.exercise_type}
              onChange={this.props.handleChange}
            />
            <br />
            <br />

            <label htmlFor="water">Ounces of water consumed</label>
            <br />
            <input
              type="text"
              name="water"
              id="water"
              value={this.state.water}
              onChange={this.props.handleChange}
            />
          </section>
        </div>
        <br />
        <br />
        <div className="log_section">
          <section className="other">
            <label htmlFor="notes">Additional notes</label>
            <br />
            <textarea
              id="notes"
              name="notes"
              rows="4"
              cols="50"
              placeholder="Anything else you would like to track? Note it here!"
              value={this.state.notes}
              onChange={this.props.handleChange}
            />
          </section>
        </div>
        <section className="button-section">
          <button type="submit">Submit</button>
          <button type="reset">Reset</button>
        </section>
      </form>
    );
  }
}

export default LogInputForm;
