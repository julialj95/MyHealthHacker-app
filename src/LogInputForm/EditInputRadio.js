import React from "react";

function EditRadioInput(props) {
  return (
    <input
      type="radio"
      id={props.id}
      name={props.type}
      value={props.value}
      checked={props.type === props.value}
      onChange={props.handleChange}
    />
  );
}

export default EditRadioInput;
