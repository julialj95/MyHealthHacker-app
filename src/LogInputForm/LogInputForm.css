@import url("https://fonts.googleapis.com/css2?family=Lato&family=Nunito&family=Source+Sans+Pro&display=swap");

.mental,
.exercise,
.sleep,
.other {
  border: solid 5px #88d317;
  padding-bottom: 10px;
  text-align: center;
  background-color: #6e3667;
  margin-left: 5%;
  margin-right: 5%;

  border-radius: 15px;
  color: white;
}

.date_section {
  padding: 0 10px 0 10px;
}
.input_box,
textarea {
  width: 60%;
}
.box_label {
  font-size: 16px;
  font-weight: normal;
}
.section_header {
  font-size: 19px;
}

.bold {
  font-weight: bold;
}

.form_btn {
  flex: 1;
  margin: 5px 10px 5px 10px;
  background-color: #3c3c3c;
  box-shadow: 1px 1px white;
  border: 2px solid #88d317;
  border-radius: 3px;
  color: white;
  font-family: "Nunito";
}
@media all and (min-width: 600px) {
  .input_box,
  textarea {
    width: 40%;
  }
  .box_label {
    font-size: 19px;
  }
  .section_header {
    font-size: 22px;
  }
  .form_btn {
    font-size: 15px;
  }
}

@media all and (min-width: 960px) {
  .input_box,
  textarea {
    width: 30%;
  }
  .box_label {
    font-size: 21px;
  }
  .section_header {
    font-size: 24px;
  }
  .form_btn {
    font-size: 18px;
  }
}
