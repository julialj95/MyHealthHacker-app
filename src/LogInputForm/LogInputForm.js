import React from "react";
import moment from "moment";
import "./LogInputForm.css";

function LogInputForm(props) {
  const {
    log_date,
    mood,
    stress,
    sleep_hours,
    sleep_quality,
    exercise_type,
    exercise_minutes,
    water,
    notes,
    handleChange,
    handleSubmitLog,
    handleReset,
  } = props;
  const today = new Date();
  const maxDate = moment(today).format("YYYY-MM-DD");
  return (
    <form onSubmit={handleSubmitLog}>
      <section className="date_section">
        <label className="box_label" htmlFor="date">
          Log Date{" "}
        </label>
        <input
          type="date"
          id="log_date"
          name="log_date"
          value={log_date}
          max={maxDate}
          onChange={handleChange}
        />
      </section>
      <br />

      <div>
        <section className="mental">
          <h2 className="section_header">Psychological</h2>
          <div>
            <h3 className="box_label bold">Mood Rating</h3>
            <h3 className="box_label">
              1 = Bad Mood <br /> 5 = Great Mood
            </h3>
            {/* {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="1"
                name="mood"
                value={1}
                onChange={handleChange}
              />
            ) : ( */}
            <input
              type="radio"
              id="1"
              name="mood"
              value={1}
              checked={mood === 1}
              onChange={handleChange}
            />
            {/* )} */}
            <label htmlFor="mood">1</label>
            {/* {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="2"
                name="mood"
                value={2}
                onChange={handleChange}
              />
            ) : ( */}
            <input
              type="radio"
              id="2"
              name="mood"
              value={2}
              checked={mood === 2}
              onChange={handleChange}
            />
            {/* // )} */}
            <label htmlFor="mood">2</label>
            {/* {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="3"
                name="mood"
                value={3}
                onChange={handleChange}
              />
            ) : ( */}
            <input
              type="radio"
              id="3"
              name="mood"
              value={3}
              checked={mood === 3}
              onChange={handleChange}
            />
            {/* )} */}
            <label htmlFor="mood">3</label>
            {/* {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="4"
                name="mood"
                value={4}
                onChange={handleChange}
              />
            ) : ( */}
            <input
              type="radio"
              id="4"
              name="mood"
              value={4}
              checked={mood === 4}
              onChange={handleChange}
            />
            {/* )} */}
            <label htmlFor="mood">4</label>
            {/* {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="5"
                name="mood"
                value={5}
                onChange={handleChange}
              />
            ) : ( */}
            <input
              type="radio"
              id="5"
              name="mood"
              value={5}
              checked={mood === 5}
              onChange={handleChange}
            />
            {/* )} */}
            <label htmlFor="mood">5</label>
          </div>
          <div className="stress">
            <h2 className="box_label bold">Stress Level</h2>
            <h3 className="box_label">
              1 = Low Stress <br />5 = High Stress
            </h3>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="1"
                name="stress"
                value={1}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="1"
                name="stress"
                value={1}
                checked={stress === 1}
                onChange={handleChange}
              />
            )}
            <label htmlFor="stress">1</label>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="2"
                name="stress"
                value={2}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="2"
                name="stress"
                value={2}
                checked={stress === 2}
                onChange={handleChange}
              />
            )}
            <label htmlFor="stress">2</label>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="3"
                name="stress"
                value={3}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="3"
                name="stress"
                value={3}
                checked={stress === 3}
                onChange={handleChange}
              />
            )}
            <label htmlFor="stress">3</label>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="4"
                name="stress"
                value={4}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="4"
                name="stress"
                value={4}
                checked={stress === 4}
                onChange={handleChange}
              />
            )}
            <label htmlFor="stress">4</label>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="5"
                name="stress"
                value={5}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="5"
                name="stress"
                value={5}
                checked={stress === 5}
                onChange={handleChange}
              />
            )}
            <label htmlFor="stress">5</label>
          </div>
        </section>
        <section className="sleep">
          <h2 className="section_header">Sleep</h2>
          <label className="box_label" htmlFor="sleep">
            Hours slept
          </label>
          <br />
          <input
            type="number"
            className="input_box"
            name="sleep_hours"
            id="sleep_hours"
            value={sleep_hours}
            onChange={handleChange}
          />
          <div>
            <h3 className="box_label">Sleep quality</h3>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="1"
                name="sleep_quality"
                value={1}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="1"
                name="sleep_quality"
                value={1}
                checked={sleep_quality === 1}
                onChange={handleChange}
              />
            )}
            <label htmlFor="sleep_quality">1</label>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="2"
                name="sleep_quality"
                value={2}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="2"
                name="sleep_quality"
                value={2}
                checked={sleep_quality === 2}
                onChange={handleChange}
              />
            )}
            <label htmlFor="sleep_quality">2</label>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="3"
                name="sleep_quality"
                value={3}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="3"
                name="sleep_quality"
                value={3}
                checked={sleep_quality === 3}
                onChange={handleChange}
              />
            )}
            <label htmlFor="sleep_quality">3</label>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="4"
                name="sleep_quality"
                value={4}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="4"
                name="sleep_quality"
                value={4}
                checked={sleep_quality === 4}
                onChange={handleChange}
              />
            )}
            <label htmlFor="sleep_quality">4</label>
            {props.parent === "submitLog" ? (
              <input
                type="radio"
                id="5"
                name="sleep_quality"
                value={5}
                onChange={handleChange}
              />
            ) : (
              <input
                type="radio"
                id="5"
                name="sleep_quality"
                value={5}
                checked={sleep_quality === 5}
                onChange={handleChange}
              />
            )}
            <label htmlFor="sleep_quality">5</label>
          </div>
        </section>

        <section className="exercise">
          <h2 className="section_header">Physical</h2>
          <label className="box_label" htmlFor="exercise_minutes">
            Minutes of exercise
          </label>
          <br />
          <input
            type="number"
            className="input_box"
            name="exercise_minutes"
            id="exercise_minutes"
            value={exercise_minutes}
            onChange={handleChange}
          />
          <br />
          <br />
          <label className="box_label" htmlFor="exercise_type">
            Type of exercise
          </label>
          <br />
          <input
            type="text"
            className="input_box"
            name="exercise_type"
            id="exercise_type"
            value={exercise_type}
            onChange={handleChange}
          />
          <br />
          <br />

          <label className="box_label" htmlFor="water">
            Water consumed (oz.)
          </label>
          <br />
          <input
            type="number"
            className="input_box"
            name="water"
            id="water"
            value={water}
            onChange={handleChange}
          />
        </section>
      </div>

      <div>
        <section className="other">
          <label className="box_label" htmlFor="notes">
            Additional notes
          </label>
          <br />
          <textarea
            id="notes"
            name="notes"
            rows="4"
            placeholder="Anything else you would like to track? Note it here!"
            value={notes}
            onChange={handleChange}
          />
        </section>
      </div>
      <section className="button-section">
        <button className="form_btn" type="submit">
          Submit
        </button>
        <button className="form_btn" type="reset" onClick={handleReset}>
          Reset
        </button>
      </section>
    </form>
  );
}

LogInputForm.defaultProps = {
  handleChange: () => {},
};

export default LogInputForm;
