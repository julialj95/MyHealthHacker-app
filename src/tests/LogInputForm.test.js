import React from "react";
import ReactDOM from "react-dom";
import LogInputForm from "../LogInputForm/LogInputForm";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <Router>
      <LogInputForm />
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <Router>
        <LogInputForm />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
