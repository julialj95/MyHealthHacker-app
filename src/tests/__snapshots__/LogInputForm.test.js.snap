// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form>
  <section
    className="date_section"
  >
    <label
      className="box_label"
      htmlFor="date"
    >
      Log Date
       
    </label>
    <input
      id="log_date"
      max="2021-05-02"
      name="log_date"
      onChange={[Function]}
      type="date"
    />
  </section>
  <br />
  <div>
    <section
      className="mental"
    >
      <h2
        className="section_header"
      >
        Psychological
      </h2>
      <div>
        <h3
          className="box_label bold"
        >
          Mood Rating
        </h3>
        <h3
          className="box_label"
        >
          1 = Bad Mood 
          <br />
           5 = Great Mood
        </h3>
        <input
          checked={false}
          id="1"
          name="mood"
          onChange={[Function]}
          type="radio"
          value={1}
        />
        <label
          htmlFor="mood"
        >
          1
        </label>
        <input
          checked={false}
          id="2"
          name="mood"
          onChange={[Function]}
          type="radio"
          value={2}
        />
        <label
          htmlFor="mood"
        >
          2
        </label>
        <input
          checked={false}
          id="3"
          name="mood"
          onChange={[Function]}
          type="radio"
          value={3}
        />
        <label
          htmlFor="mood"
        >
          3
        </label>
        <input
          checked={false}
          id="4"
          name="mood"
          onChange={[Function]}
          type="radio"
          value={4}
        />
        <label
          htmlFor="mood"
        >
          4
        </label>
        <input
          checked={false}
          id="5"
          name="mood"
          onChange={[Function]}
          type="radio"
          value={5}
        />
        <label
          htmlFor="mood"
        >
          5
        </label>
      </div>
      <div
        className="stress"
      >
        <h2
          className="box_label bold"
        >
          Stress Level
        </h2>
        <h3
          className="box_label"
        >
          1 = Low Stress 
          <br />
          5 = High Stress
        </h3>
        <input
          checked={false}
          id="1"
          name="stress"
          onChange={[Function]}
          type="radio"
          value={1}
        />
        <label
          htmlFor="stress"
        >
          1
        </label>
        <input
          checked={false}
          id="2"
          name="stress"
          onChange={[Function]}
          type="radio"
          value={2}
        />
        <label
          htmlFor="stress"
        >
          2
        </label>
        <input
          checked={false}
          id="3"
          name="stress"
          onChange={[Function]}
          type="radio"
          value={3}
        />
        <label
          htmlFor="stress"
        >
          3
        </label>
        <input
          checked={false}
          id="4"
          name="stress"
          onChange={[Function]}
          type="radio"
          value={4}
        />
        <label
          htmlFor="stress"
        >
          4
        </label>
        <input
          checked={false}
          id="5"
          name="stress"
          onChange={[Function]}
          type="radio"
          value={5}
        />
        <label
          htmlFor="stress"
        >
          5
        </label>
      </div>
    </section>
    <section
      className="sleep"
    >
      <h2
        className="section_header"
      >
        Sleep
      </h2>
      <label
        className="box_label"
        htmlFor="sleep"
      >
        Hours slept
      </label>
      <br />
      <input
        className="input_box"
        id="sleep_hours"
        name="sleep_hours"
        onChange={[Function]}
        type="number"
      />
      <div>
        <h3
          className="box_label"
        >
          Sleep quality
        </h3>
        <input
          checked={false}
          id="1"
          name="sleep_quality"
          onChange={[Function]}
          type="radio"
          value={1}
        />
        <label
          htmlFor="sleep_quality"
        >
          1
        </label>
        <input
          checked={false}
          id="2"
          name="sleep_quality"
          onChange={[Function]}
          type="radio"
          value={2}
        />
        <label
          htmlFor="sleep_quality"
        >
          2
        </label>
        <input
          checked={false}
          id="3"
          name="sleep_quality"
          onChange={[Function]}
          type="radio"
          value={3}
        />
        <label
          htmlFor="sleep_quality"
        >
          3
        </label>
        <input
          checked={false}
          id="4"
          name="sleep_quality"
          onChange={[Function]}
          type="radio"
          value={4}
        />
        <label
          htmlFor="sleep_quality"
        >
          4
        </label>
        <input
          checked={false}
          id="5"
          name="sleep_quality"
          onChange={[Function]}
          type="radio"
          value={5}
        />
        <label
          htmlFor="sleep_quality"
        >
          5
        </label>
      </div>
    </section>
    <section
      className="exercise"
    >
      <h2
        className="section_header"
      >
        Physical
      </h2>
      <label
        className="box_label"
        htmlFor="exercise_minutes"
      >
        Minutes of exercise
      </label>
      <br />
      <input
        className="input_box"
        id="exercise_minutes"
        name="exercise_minutes"
        onChange={[Function]}
        type="number"
      />
      <br />
      <br />
      <label
        className="box_label"
        htmlFor="exercise_type"
      >
        Type of exercise
      </label>
      <br />
      <input
        className="input_box"
        id="exercise_type"
        name="exercise_type"
        onChange={[Function]}
        type="text"
      />
      <br />
      <br />
      <label
        className="box_label"
        htmlFor="water"
      >
        Water consumed (oz.)
      </label>
      <br />
      <input
        className="input_box"
        id="water"
        name="water"
        onChange={[Function]}
        type="number"
      />
    </section>
  </div>
  <div>
    <section
      className="other"
    >
      <label
        className="box_label"
        htmlFor="notes"
      >
        Additional notes
      </label>
      <br />
      <textarea
        id="notes"
        name="notes"
        onChange={[Function]}
        placeholder="Anything else you would like to track? Note it here!"
        rows="4"
      />
    </section>
  </div>
  <section
    className="button-section"
  >
    <button
      className="form_btn"
      type="submit"
    >
      Submit
    </button>
    <button
      className="form_btn"
      type="reset"
    >
      Reset
    </button>
  </section>
</form>
`;
